### ISSUE TITLE
How to build without nvidia-cg-toolkit on Linux?

### ISSUE THREAD

**Issue by ddevault** [PCSX2/pcsx2#2904]
I'm trying to avoid a nonfree PCSX2 build.

**Comment by arcum42** [comment]
Not building the extra plugins would be the surest way. 

The only plugins that use it are zerogs and zz-ogl, though. zz-ogl won't use it if you build with glsl instead. Zerogs only builds with the nvidia-cg-toolkit, but it doesn't normally build anyways, IIRC.

So pretty much either: sh build.sh 
or: sh build.sh --extra --glsl

**Comment by lightningterror** [comment]
I'll close this since it's not a valid issue and arcum answered the question.

**Comment by ddevault** [comment]
Thanks!

**Comment by arcum42** [comment]
No problem. Strictly speaking, everyone uses GSdx instead of either of those plugins anyways, so unless you have a particular reason to use a graphics plugin that hasn't actually been worked on for years, skipping both shouldn't be a big deal.

**Comment by ddevault** [comment]
If it's abandoned and obsolete, should it be removed?

**Comment by arcum42** [comment]
Well, it's a different implementation, which could be useful for comparison, potentially, and it does work, just not as well as GSdx. It's in extra plugins for a reason, though, and removing it'll undoubtedly be discussed at some point. There are other plugins even more in need of removal, though. (And I've removed a few of those already...)

I'm also the last person that did any work on zzogl, and a certain part of me hates deleting a plugin I put a bunch of work into, too, if you know what I mean...

**Comment by ddevault** [comment]
Yeah, I get it, been there before myself. I'd personally think about slating this for speedy removal though, on the basis that it requires nonfree software and is no longer necessary.

**Comment by arcum42** [comment]
It is free software, actually, just not open source. You can download it here:
https://developer.nvidia.com/cg-toolkit-download

I could tweak things to make zzogl use glsl by default easily enough, in any case.

**Comment by gregory38** [comment]
I'm against the removal. You can still find some useful info into those old GS plugins. Besides glsl port is easy (if not already done). I don't remember the exact status but I didn't do the windows side (which should be very easy) and potentially I hit some AMD driver bugs (that hopefully should be fixed now)

Anyway extra plugins are for developers not for standard users.

**Comment by ddevault** [comment]
Free as in freedom, I meant. Anyway, it's your call. Thanks!

**Comment by arcum42** [comment]
np.

I recall glsl working fine, though there could easily be bugs I didn't run into. ZZOgl didn't compile on Windows, last time I tried.

I wouldn't mind making glsl the default on it, and either having cg as optional, or even removing it. I largely haven't been doing things on zzogl in trunk because I figured no one else cares about the plugin, and there are enough other things to work on. (Not to mention I only really seem to be coding once every week or two lately...)

I've got better screenshot code around for it somewhere, in fact, that adds another format or two, and fixes things about what's currently in there for formats...
